### PR TITLE
Bottom sheet docs

### PR DESCRIPTION
**Pull request by clocksmith**
Add bottom sheets doc. This version is not ready to be published yet because the Expanding bottom sheet section should link to the Expanding bottom sheet page

### CODE REVIEW

File: docs/components/bottom_sheets.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+### Making bottom sheets accessible
+
+Flutter's APIs support accessibility setting for large fonts, screen readers, and sufficient contrast. For more information, go to Flutter's [accessibility](https://flutter.dev/docs/development/accessibility-and-localization/accessibility) and [internationalization](https://flutter.dev/docs/development/accessibility-and-localization/internationalization) pages.
+
+For more guidance on writing labels, go to [our page on how to write a good accessibility label](https://material.io/design/usability/accessibility.html#writing).
+
+## Types
+
+There are three types suitable for different use cases:
```
Comment (JoseAlba) [inline docs/components/bottom_sheets.md:47]: empty space

File: docs/components/bottom_sheets.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+1. [Standard bottom sheets](#standard-bottom-sheet) display content that complements the screen’s primary content and remain visible while users interact with the primary content.
```
Comment (JoseAlba) [inline docs/components/bottom_sheets.md:48]: 1.2.3 => 1.1.1.

File: docs/components/bottom_sheets.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+![Composite image of bottom sheet types](assets/bottom_sheets/bottom-sheet_types_composite.png)
+
+## Standard bottom sheet
+
+Standard bottom sheets co-exist with the screen’s main UI region and allow for simultaneously viewing and interacting with both regions. They are commonly used to keep a feature or secondary content visible on screen when content in main UI region is frequently scrolled or panned.
+
+### Standard bottom sheet example
+
+`BottomSheet`
```
Comment (JoseAlba) [inline docs/components/bottom_sheets.md:60]: empty space after bottom sheet

File: docs/components/bottom_sheets.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+&nbsp;         | Property
+-------------- | ------------------------
+| **Color** | `barrierColor` |
+
+
+## Expanding bottom sheet
+
+Expanding bottom sheets require creating a custom widget. See [Expanding bottom sheet](link) for more info.
+TODO: link https://github.com/material-components/material-components-flutter/blob/develop/docs/components/expanding-bottom-sheet.md on material.io
```
Comment (JoseAlba) [inline docs/components/bottom_sheets.md:272]: I don't know about having the TODO

File: docs/components/dartpad/bottom_sheets/modal/dartpad_metadata.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+name: Material Modal Bottom Sheet Demonstration
+mode: flutter
+files:
+  - name: main.dart
```
Comment (JoseAlba) [inline docs/components/dartpad/bottom_sheets/modal/dartpad_metadata.yaml:4]: empty line

File: docs/components/dartpad/bottom_sheets/modal/main.dart [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                  );
+                },
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
```
Comment (JoseAlba) [inline docs/components/dartpad/bottom_sheets/modal/main.dart:55]: empty line

File: docs/components/dartpad/bottom_sheets/standard/dartpad_metadata.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+name: Material Standard Bottom Sheet Demonstration
+mode: flutter
+files:
+  - name: main.dart
```
Comment (JoseAlba) [inline docs/components/dartpad/bottom_sheets/standard/dartpad_metadata.yaml:4]: empty line

File: docs/components/dartpad/bottom_sheets/standard/main.dart [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+                  );
+                },
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
```
Comment (JoseAlba) [inline docs/components/dartpad/bottom_sheets/standard/main.dart:69]: empty line

**Review by JoseAlba — APPROVED** [review]
Small problems but LGTM

File: docs/components/bottom_sheets.md [hunk 9]
```diff
@@ -0,0 +1,280 @@
...
+        colorScheme: Theme.of(context).colorScheme.copyWith(
+          primary: Color(0xFF6200EE),
+        ),
+      ),
+      home: Scaffold(
+        // Use a Builder to get a context within the Scaffold.
+        body: Builder(
+          builder: (context) {
+            return Center(
+              child: FlatButton(
```
Comment (rami-a) [inline docs/components/bottom_sheets.md:92]: Use the new button universe here and in the dartpad demos

File: docs/components/bottom_sheets.md [hunk 10]
```diff
@@ -0,0 +1,280 @@
...
+                child: Text('Show Bottom Sheet'),
```
Comment (rami-a) [inline docs/components/bottom_sheets.md:93]: Upper case the button text here and in the dartpad demos

File: docs/components/bottom_sheets.md [hunk 11]
```diff
@@ -0,0 +1,280 @@
...
+## Modal bottom sheet
+
+Modal bottom sheets present a set of choices while blocking interaction with the rest of the screen. They are an alternative to inline menus and simple dialogs on mobile, providing additional room for content, iconography, and actions.
+
+Modal bottom sheets are used in mobile apps only.
+
+### Modal bottom sheet example
+
+`BottomSheet`
+- [Class definition](https://api.flutter.dev/flutter/material/BottomSheet-class.html)
```
Comment (rami-a) [inline docs/components/bottom_sheets.md:174]: This should probably link to the showModalBottomSheet helper instead

File: docs/components/bottom_sheets.md [hunk 12]
```diff
@@ -0,0 +1,280 @@
...
+
+
+## Expanding bottom sheet
+
+Expanding bottom sheets require creating a custom widget. See [Expanding bottom sheet](link) for more info.
+TODO: link https://github.com/material-components/material-components-flutter/blob/develop/docs/components/expanding-bottom-sheet.md on material.io
+
+## Theming bottom sheets
+
+Theming for bottom sheets can be done by theming the widgets that are inside the `widget` returned by the `builder` of `showBottomSheet` or `showModalBottomSheet`.
```
Comment (rami-a) [inline docs/components/bottom_sheets.md:276]: There is also a bottomsheettheme object i think

**Review by rami-a — APPROVED** [review]
LGTM but a bunch of the assets aren't being used I think and should be removed

File: docs/components/bottom_sheets.md (hunk 11 above)
Comment (clocksmith) [reply docs/components/bottom_sheets.md]: Good idea

File: docs/components/bottom_sheets.md (hunk 12 above)
Comment (clocksmith) [reply docs/components/bottom_sheets.md]: Ah there is, added details on it

File: docs/components/bottom_sheets.md [hunk 13]
```diff
@@ -0,0 +1,286 @@
...
+excerpt: "Bottom sheets are surfaces containing supplementary content that are anchored to the bottom of the screen."
+iconId:
+path: /catalog/material-sheets-bottom/
+-->
+
+# Sheets: bottom
+
+[Bottom sheets](https://material.io/components/sheets-bottom) are surfaces containing supplementary content that are anchored to the bottom of the screen.
+
+![Example bottom sheet: modal bottom sheet](assets/bottom_sheets/SheetsBottom_hero.png)
```
Comment (mingjane-work) [inline docs/components/bottom_sheets.md:14]: replace alt-text with:

Bottom sheet with 4 items, with scrim covering the remaining screen.

File: docs/components/bottom_sheets.md [hunk 14]
```diff
@@ -0,0 +1,286 @@
...
+
+For more guidance on writing labels, go to [our page on how to write a good accessibility label](https://material.io/design/usability/accessibility.html#writing).
+
+## Types
+
+There are three types suitable for different use cases:
+1. [Standard bottom sheets](#standard-bottom-sheet) display content that complements the screen’s primary content and remain visible while users interact with the primary content.
+1. [Modal bottom sheets](#modal-bottom-sheet) are an alternative to inline menus or simple dialogs on mobile and provide room for additional items, longer descriptions, and iconography, and must be dismissed in order to interact with the underlying content.
+1. [Expanding bottom sheets](#expanding-bottom-sheet) provide a small, collapsed surface that can be expanded by the user to access a key feature or task to offer the persistent access of a standard sheet with the space and focus of a modal sheet.
```
Comment (mingjane-work) [inline docs/components/bottom_sheets.md:50]: The descriptive content for each type can move to the type's section. The remaining list/links to type sections should be an inline list:

"There are three types of bottom sheets: 1\. standard... 2\. ... 3\. ..."

File: docs/components/bottom_sheets.md [hunk 15]
```diff
@@ -0,0 +1,286 @@
...
+
+`BottomSheet`
+
+- [Class definition](https://api.flutter.dev/flutter/material/ScaffoldState/showBottomSheet.html)
+- [GitHub source](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/bottom_sheet.dart)
+- [Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/bottom_sheets/standard&gh_ref=develop)
+
+
+The following is an example expanded standard bottom sheet:
+![An expanded standard bottom sheet with header and list items](assets/bottom_sheets/SheetsBottom_standard_screenshot.jpg)
```
Comment (mingjane-work) [inline docs/components/bottom_sheets.md:68]: Suggested alt-text:

Bottom sheet with "Header" and 5 items partially covering image of paper stacks.

File: docs/components/bottom_sheets.md [hunk 16]
```diff
@@ -0,0 +1,286 @@
...
+
+### Modal bottom sheet example
+
+`BottomSheet`
+- [Class definition](https://api.flutter.dev/flutter/material/showModalBottomSheet.html)
+- [GitHub source](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/bottom_sheet.dart)
+- [Dartpad demo](https://dartpad.dev/embed-flutter.html?gh_owner=material-components&gh_repo=material-components-flutter&gh_path=docs/components/dartpad/bottom_sheets/modal&gh_ref=develop)
+
+The following is an example modal bottom sheet:
+![A modal bottom sheet with share, link, edit, and delete list items](assets/bottom_sheets/SheetsBottom_modal_screenshot.jpg)
```
Comment (mingjane-work) [inline docs/components/bottom_sheets.md:180]: Suggested alt-text:

Bottom sheet with share, link, edit, and delete items partially covering screen image with scrim of dogs drinking water.

File: docs/components/bottom_sheets.md (hunk 4 above)
Comment (mingjane-work) [reply docs/components/bottom_sheets.md]: Go ahead and link to the the github page.

File: docs/components/bottom_sheets.md [hunk 17]
```diff
@@ -0,0 +1,286 @@
...
+
+
+## Expanding bottom sheet
+
+Expanding bottom sheets require creating a custom widget. See [Expanding bottom sheet](link) for more info.
+TODO: link https://github.com/material-components/material-components-flutter/blob/develop/docs/components/expanding-bottom-sheet.md on material.io
+
+## Theming bottom sheets
+
+Bottom Sheets support [Material Theming](https://material.io/components/sheets-bottom#theming) and can be customized in terms of color, elevation and shape.
```
Comment (mingjane-work) [inline docs/components/bottom_sheets.md:276]: Why isn't there example code/screenshot?

Comment (clocksmith) [reply docs/components/bottom_sheets.md:276]: The shrine study doesn't show what to do for the shape or elevation or background color. Applying the shrine theme as we do in other components makes no change, so the only change here would be to modify the BottomSheetThemeData with some shape, elevation, or color.

Comment (mingjane-work) [reply docs/components/bottom_sheets.md:275]: Though Shrine text is brown, and the standard sheet would have a pink header with brown text. Can we switch to using the standard example except in Shrine?

Comment (clocksmith) [reply docs/components/bottom_sheets.md:275]: Ah good point, updated with standard example in shrine theme

**Review by mingjane-work — APPROVED** [review]
LGTM

**Pull request merged**